File: drestaurant-libs/drestaurant-courier/src/test/java/com/drestaurant/courier/domain/CourierOrderAggregateTest.java
package com.drestaurant.courier.domain;

import org.axonframework.messaging.interceptors.BeanValidationInterceptor;
import org.axonframework.test.aggregate.AggregateTestFixture;
import org.axonframework.test.aggregate.FixtureConfiguration;
import org.junit.Before;
import org.junit.Test;

import com.drestaurant.common.domain.model.AuditEntry;
import com.drestaurant.courier.domain.api.AssignCourierOrderToCourierCommand;
import com.drestaurant.courier.domain.api.CourierOrderAssignedEvent;
import com.drestaurant.courier.domain.api.CourierOrderCreatedEvent;
import com.drestaurant.courier.domain.api.CourierOrderDeliveredEvent;
import com.drestaurant.courier.domain.api.CreateCourierOrderCommand;
import com.drestaurant.courier.domain.api.MarkCourierOrderAsDeliveredCommand;

public class CourierOrderAggregateTest {

	private FixtureConfiguration<CourierOrder> fixture;
	private AuditEntry auditEntry;
	private static final String WHO = "johndoe";
	
	private String orderId;
	private String courierId;

	@Before
	public void setUp() throws Exception {
		fixture = new AggregateTestFixture<>(CourierOrder.class);
		fixture.registerCommandDispatchInterceptor(new BeanValidationInterceptor<>());
		auditEntry = new AuditEntry(WHO);
		orderId = "orderId";
		courierId = "courierId";
	}
	
	@Test
	public void createCourierOrderTest() throws Exception {
		CreateCourierOrderCommand createCourierOrderCommand = new CreateCourierOrderCommand(orderId, auditEntry);
		CourierOrderCreatedEvent courierOrderCreatedEvent = new CourierOrderCreatedEvent( orderId, auditEntry);

		fixture
		.given()
		.when(createCourierOrderCommand)
		.expectEvents(courierOrderCreatedEvent);
	}

	@Test
	public void assignOdrerToCourierTest() throws Exception {
		CourierOrderCreatedEvent courierOrderCreatedEvent = new CourierOrderCreatedEvent(orderId, auditEntry);
		AssignCourierOrderToCourierCommand assignCourierOrderToCourierCommand = new AssignCourierOrderToCourierCommand(orderId, courierId, auditEntry);
		CourierOrderAssigningInitiatedEvent courierOrderAssigningInitiatedEvent = new CourierOrderAssigningInitiatedEvent(courierId, orderId, auditEntry);

		fixture
		.given(courierOrderCreatedEvent)
		.when(assignCourierOrderToCourierCommand)
		.expectEvents(courierOrderAssigningInitiatedEvent);
	}
	
	@Test
	public void markOrderAsAssignedTest() throws Exception {
		CourierOrderCreatedEvent courierOrderCreatedEvent = new CourierOrderCreatedEvent(orderId, auditEntry);
		MarkCourierOrderAsAssignedCommand markCourierOrderAsAssignedCommand = new MarkCourierOrderAsAssignedCommand(orderId, courierId, auditEntry);
		CourierOrderAssigningInitiatedEvent courierOrderAssigningInitiatedEvent = new CourierOrderAssigningInitiatedEvent(courierId, orderId, auditEntry);
		CourierOrderAssignedEvent courierOrderAssignedEvent = new CourierOrderAssignedEvent(orderId, courierId, auditEntry);
		
		fixture
		.given(courierOrderCreatedEvent, courierOrderAssigningInitiatedEvent)
		.when(markCourierOrderAsAssignedCommand)
		.expectEvents(courierOrderAssignedEvent);
	}
	
	@Test
	public void markOrderAsNotAssignedTest() throws Exception {
		CourierOrderCreatedEvent courierOrderCreatedEvent = new CourierOrderCreatedEvent(orderId, auditEntry);
		MarkCourierOrderAsNotAssignedCommand markCourierOrderAsNotAssignedCommand = new MarkCourierOrderAsNotAssignedCommand(orderId, auditEntry);
		CourierOrderAssigningInitiatedEvent courierOrderAssigningInitiatedEvent = new CourierOrderAssigningInitiatedEvent(courierId, orderId, auditEntry);
		CourierOrderNotAssignedEvent courierOrderNotAssignedEvent = new CourierOrderNotAssignedEvent(orderId, auditEntry);
		
		fixture
		.given(courierOrderCreatedEvent, courierOrderAssigningInitiatedEvent)
		.when(markCourierOrderAsNotAssignedCommand)
		.expectEvents(courierOrderNotAssignedEvent);
	}
	
	@Test
	public void markOrderAsDeliveredTest() throws Exception {
		CourierOrderCreatedEvent courierOrderCreatedEvent = new CourierOrderCreatedEvent(orderId, auditEntry);
		CourierOrderAssigningInitiatedEvent courierOrderAssigningInitiatedEvent = new CourierOrderAssigningInitiatedEvent(courierId, orderId, auditEntry);
		CourierOrderAssignedEvent courierOrderAssignedEvent = new CourierOrderAssignedEvent(orderId, courierId, auditEntry);
		MarkCourierOrderAsDeliveredCommand markCourierOrderAsDeliveredCommand = new MarkCourierOrderAsDeliveredCommand(orderId, auditEntry); 
		CourierOrderDeliveredEvent courierOrderDeliveredEvent = new CourierOrderDeliveredEvent(orderId, auditEntry);
		fixture
		.given(courierOrderCreatedEvent, courierOrderAssigningInitiatedEvent, courierOrderAssignedEvent)
		.when(markCourierOrderAsDeliveredCommand)
		.expectEvents(courierOrderDeliveredEvent);
	}

}

File: drestaurant-libs/drestaurant-courier/src/test/kotlin/com/drestaurant/courier/domain/CourierOrderAggregateTest.kt
package com.drestaurant.courier.domain

import com.drestaurant.common.domain.api.model.AuditEntry
import com.drestaurant.courier.domain.api.*
import com.drestaurant.courier.domain.api.model.CourierId
import com.drestaurant.courier.domain.api.model.CourierOrderId
import org.axonframework.messaging.interceptors.BeanValidationInterceptor
import org.axonframework.test.aggregate.AggregateTestFixture
import org.axonframework.test.aggregate.FixtureConfiguration
import org.junit.Before
import org.junit.Test
import java.util.*

class CourierOrderAggregateTest {

    private lateinit var fixture: FixtureConfiguration<CourierOrder>
    private lateinit var auditEntry: AuditEntry

    private lateinit var orderId: CourierOrderId
    private lateinit var courierId: CourierId

    private val who = "johndoe"

    @Before
    fun setUp() {
        fixture = AggregateTestFixture(CourierOrder::class.java)
        fixture.registerCommandDispatchInterceptor(BeanValidationInterceptor())
        auditEntry = AuditEntry(who, Calendar.getInstance().time)
        orderId = CourierOrderId("orderId")
        courierId = CourierId("courierId")
    }

    @Test
    fun createCourierOrderTest() {
        val createCourierOrderCommand = CreateCourierOrderCommand(orderId, auditEntry)
        val courierOrderCreatedEvent = CourierOrderCreatedEvent(orderId, auditEntry)

        fixture
                .given()
                .`when`(createCourierOrderCommand)
                .expectEvents(courierOrderCreatedEvent)
    }

    @Test
    fun assignOdrerToCourierTest() {
        val courierOrderCreatedEvent = CourierOrderCreatedEvent(orderId, auditEntry)
        val assignCourierOrderToCourierCommand = AssignCourierOrderToCourierCommand(orderId, courierId, auditEntry)
        val courierOrderAssigningInitiatedEvent = CourierOrderAssigningInitiatedInternalEvent(courierId, orderId, auditEntry)

        fixture
                .given(courierOrderCreatedEvent)
                .`when`(assignCourierOrderToCourierCommand)
                .expectEvents(courierOrderAssigningInitiatedEvent)
    }

    @Test
    fun markOrderAsAssignedTest() {
        val courierOrderCreatedEvent = CourierOrderCreatedEvent(orderId, auditEntry)
        val markCourierOrderAsAssignedCommand = MarkCourierOrderAsAssignedInternalCommand(orderId, courierId, auditEntry)
        val courierOrderAssigningInitiatedEvent = CourierOrderAssigningInitiatedInternalEvent(courierId, orderId, auditEntry)
        val courierOrderAssignedEvent = CourierOrderAssignedEvent(orderId, courierId, auditEntry)

        fixture
                .given(courierOrderCreatedEvent, courierOrderAssigningInitiatedEvent)
                .`when`(markCourierOrderAsAssignedCommand)
                .expectEvents(courierOrderAssignedEvent)
    }

    @Test
    fun markOrderAsNotAssignedTest() {
        val courierOrderCreatedEvent = CourierOrderCreatedEvent(orderId, auditEntry)
        val markCourierOrderAsNotAssignedCommand = MarkCourierOrderAsNotAssignedInternalCommand(orderId, auditEntry)
        val courierOrderAssigningInitiatedEvent = CourierOrderAssigningInitiatedInternalEvent(courierId, orderId, auditEntry)
        val courierOrderNotAssignedEvent = CourierOrderNotAssignedEvent(orderId, auditEntry)

        fixture
                .given(courierOrderCreatedEvent, courierOrderAssigningInitiatedEvent)
                .`when`(markCourierOrderAsNotAssignedCommand)
                .expectEvents(courierOrderNotAssignedEvent)
    }

    @Test
    fun markOrderAsDeliveredTest() {
        val courierOrderCreatedEvent = CourierOrderCreatedEvent(orderId, auditEntry)
        val courierOrderAssigningInitiatedEvent = CourierOrderAssigningInitiatedInternalEvent(courierId, orderId, auditEntry)
        val courierOrderAssignedEvent = CourierOrderAssignedEvent(orderId, courierId, auditEntry)
        val markCourierOrderAsDeliveredCommand = MarkCourierOrderAsDeliveredCommand(orderId, auditEntry)
        val courierOrderDeliveredEvent = CourierOrderDeliveredEvent(orderId, auditEntry)
        fixture
                .given(courierOrderCreatedEvent, courierOrderAssigningInitiatedEvent, courierOrderAssignedEvent)
                .`when`(markCourierOrderAsDeliveredCommand)
                .expectEvents(courierOrderDeliveredEvent)
    }

}
